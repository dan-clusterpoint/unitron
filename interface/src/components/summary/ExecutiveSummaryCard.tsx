import CompanyProfileCard, { type CompanyProfileProps } from './CompanyProfileCard'
import DigitalScoreBar from './DigitalScoreBar'
import GrowthTriggersList, { type GrowthTriggersListProps } from './GrowthTriggersList'

export interface ExecutiveSummaryCardProps {
  profile: CompanyProfileProps
  score: number
  vendors: string[]
  triggers?: GrowthTriggersListProps['triggers']
  seo?: {
    robotsTxt?: boolean
    sitemapXml?: boolean
  }
}

export default function ExecutiveSummaryCard({
  profile,
  score,
  vendors,
  triggers = [],
  seo,
}: ExecutiveSummaryCardProps) {
  return (
    <section aria-labelledby="exec-summary-heading">
      <h2 id="exec-summary-heading" className="sr-only">
        Executive Summary
      </h2>
      <article className="grid grid-cols-1 xs:grid-cols-2 gap-4 p-4 bg-white rounded shadow max-h-[320px] overflow-auto">
        <div className="xs:col-span-2">
          <CompanyProfileCard {...profile} />
        </div>
        <DigitalScoreBar score={score} />
        {seo && (seo.robotsTxt !== undefined || seo.sitemapXml !== undefined) && (
          <div className="xs:col-span-2 text-xs flex gap-4" data-testid="seo-basics">
            <span className="font-medium">SEO basics:</span>
            {seo.robotsTxt !== undefined && (
              <span>Robots.txt {seo.robotsTxt ? '✓' : '✗'}</span>
            )}
            {seo.sitemapXml !== undefined && (
              <span>Sitemap {seo.sitemapXml ? '✓' : '✗'}</span>
            )}
          </div>
        )}
        {vendors.length > 0 && (
          <div className="xs:col-span-2 space-y-1 text-sm">
            {vendors.map((v) => (
              <div key={v}>{v}</div>
            ))}
          </div>
        )}
        {triggers.length > 0 && (
          <div className="xs:col-span-2">
            <GrowthTriggersList triggers={triggers} />
          </div>
        )}
      </article>
    </section>
  )
}
