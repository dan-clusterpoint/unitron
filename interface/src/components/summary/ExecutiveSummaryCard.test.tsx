import { render, screen } from '@testing-library/react'
import { test, expect } from 'vitest'
import ExecutiveSummaryCard from './ExecutiveSummaryCard'

test('renders executive summary snapshot', () => {
  const { asFragment } = render(
    <ExecutiveSummaryCard
      profile={{
        name: 'Acme Inc',
        industry: 'SaaS',
        location: 'NYC',
        website: 'https://acme.com',
        logoUrl: 'https://logo.example.com/acme.png',
      }}
      score={75}
      risk={{ x: 1, y: 2, level: 'high' }}
      stack={[
        { label: 'React', status: 'added' },
        { label: 'Vue', status: 'removed' },
      ]}
      triggers={['Add chat widget', 'Improve SEO']}
      actions={[
        { label: 'Analyze stack', targetId: 'stack' },
        { label: 'Check risks', targetId: 'risk' },
      ]}
    />,
  )
  expect(screen.getByAltText('Acme Inc logo')).toBeInTheDocument()
  expect(asFragment()).toMatchSnapshot()
})

test('omits sections when data is missing', () => {
  const {
    asFragment,
    queryByLabelText,
    queryAllByTestId,
    container,
  } = render(
    <ExecutiveSummaryCard
      profile={{
        name: 'Acme Inc',
        industry: 'SaaS',
        location: 'NYC',
        website: 'https://acme.com',
        logoUrl: 'https://logo.example.com/acme.png',
      }}
      score={75}
      stack={[]}
      triggers={[]}
      actions={[]}
    />,
  )
  expect(queryByLabelText('healthchecks')).toBeNull()
  expect(container.querySelector('.xs\\:col-span-2.space-y-1')).toBeNull()
  expect(queryAllByTestId('growth-trigger')).toHaveLength(0)
  expect(container.querySelector('ul.flex.flex-wrap.gap-2')).toBeNull()
  expect(asFragment()).toMatchSnapshot()
})
