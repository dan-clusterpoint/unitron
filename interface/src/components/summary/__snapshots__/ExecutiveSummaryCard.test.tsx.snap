// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`omits sections when data is missing 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="exec-summary-heading"
  >
    <h2
      class="sr-only"
      id="exec-summary-heading"
    >
      Executive Summary
    </h2>
    <article
      class="grid grid-cols-1 xs:grid-cols-2 gap-4 p-4 bg-white rounded shadow max-h-[320px] overflow-auto"
    >
      <div
        class="xs:col-span-2"
      >
        <div
          class="flex items-center gap-4 p-4 border rounded bg-white"
        >
          <img
            alt="Acme Inc logo"
            class="w-12 h-12 rounded object-cover"
            src="https://logo.example.com/acme.png"
          />
          <div
            class="text-sm"
          >
            <div
              class="font-medium"
            >
              Acme Inc
            </div>
            <div>
              SaaS
            </div>
            <div>
              NYC
            </div>
            <a
              class="text-blue-600 underline break-all"
              href="https://acme.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://acme.com
            </a>
          </div>
        </div>
      </div>
      <div>
        <div
          class="text-sm mb-1"
        >
          Digital Score: 75
        </div>
        <div
          class="h-2 bg-gray-200 rounded"
        >
          <div
            class="h-full rounded"
            style="width: 75%; background-color: rgb(230, 159, 0);"
          />
        </div>
      </div>
    </article>
  </section>
</DocumentFragment>
`;

exports[`renders executive summary snapshot 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="exec-summary-heading"
  >
    <h2
      class="sr-only"
      id="exec-summary-heading"
    >
      Executive Summary
    </h2>
    <article
      class="grid grid-cols-1 xs:grid-cols-2 gap-4 p-4 bg-white rounded shadow max-h-[320px] overflow-auto"
    >
      <div
        class="xs:col-span-2"
      >
        <div
          class="flex items-center gap-4 p-4 border rounded bg-white"
        >
          <img
            alt="Acme Inc logo"
            class="w-12 h-12 rounded object-cover"
            src="https://logo.example.com/acme.png"
          />
          <div
            class="text-sm"
          >
            <div
              class="font-medium"
            >
              Acme Inc
            </div>
            <div>
              SaaS
            </div>
            <div>
              NYC
            </div>
            <a
              class="text-blue-600 underline break-all"
              href="https://acme.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              https://acme.com
            </a>
          </div>
        </div>
      </div>
      <div>
        <div
          class="text-sm mb-1"
        >
          Digital Score: 75
        </div>
        <div
          class="h-2 bg-gray-200 rounded"
        >
          <div
            class="h-full rounded"
            style="width: 75%; background-color: rgb(230, 159, 0);"
          />
        </div>
      </div>
      <button
        aria-label="healthchecks"
        class="grid grid-cols-3 grid-rows-3 gap-0.5"
      >
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(213, 94, 0);"
        />
        <div
          class="w-3 h-3 border"
          style="background-color: rgb(243, 244, 246);"
        />
      </button>
      <div
        class="xs:col-span-2 space-y-1"
      >
        <div
          class="flex items-center gap-2 text-sm"
        >
          <svg
            aria-hidden="true"
            class="w-5 h-5"
            fill="currentColor"
            style="color: rgb(0, 158, 115);"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12 2.25c-5.385 0-9.75 4.365-9.75 9.75s4.365 9.75 9.75 9.75 9.75-4.365 9.75-9.75S17.385 2.25 12 2.25zM12.75 9a.75.75 0 00-1.5 0v2.25H9a.75.75 0 000 1.5h2.25V15a.75.75 0 001.5 0v-2.25H15a.75.75 0 000-1.5h-2.25V9z"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            React
          </span>
        </div>
        <div
          class="flex items-center gap-2 text-sm"
        >
          <svg
            aria-hidden="true"
            class="w-5 h-5"
            fill="currentColor"
            style="color: rgb(213, 94, 0);"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12 2.25c-5.385 0-9.75 4.365-9.75 9.75s4.365 9.75 9.75 9.75 9.75-4.365 9.75-9.75S17.385 2.25 12 2.25zm3 10.5a.75.75 0 000-1.5H9a.75.75 0 000 1.5h6z"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            Vue
          </span>
        </div>
      </div>
      <div
        class="xs:col-span-2"
      >
        <div
          class="hidden xs:block text-sm"
        >
          <ul
            class="list-disc ml-4 space-y-1"
            role="list"
          >
            <li
              data-testid="growth-trigger"
              tabindex="0"
            >
              Add chat widget
            </li>
            <li
              data-testid="growth-trigger"
              tabindex="0"
            >
              Improve SEO
            </li>
          </ul>
        </div>
        <div
          class="block xs:hidden text-sm"
          data-orientation="vertical"
        >
          <div
            class="border-b"
            data-orientation="vertical"
            data-state="closed"
          >
            <h3
              class="flex"
              data-orientation="vertical"
              data-state="closed"
            >
              <button
                aria-controls="radix-«r1»"
                aria-expanded="false"
                class="flex flex-1 items-center justify-between py-4 font-medium hover:underline"
                data-orientation="vertical"
                data-radix-collection-item=""
                data-state="closed"
                id="radix-«r0»"
                type="button"
              >
                Growth Triggers
                <svg
                  class="h-4 w-4 shrink-0 transition-transform duration-200"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </h3>
            <div
              aria-labelledby="radix-«r0»"
              class="overflow-hidden data-[state=open]:animate-accordion-down"
              data-orientation="vertical"
              data-state="closed"
              hidden=""
              id="radix-«r1»"
              role="region"
              style="--radix-accordion-content-height: var(--radix-collapsible-content-height); --radix-accordion-content-width: var(--radix-collapsible-content-width);"
            />
          </div>
        </div>
      </div>
      <div
        class="xs:col-span-2"
      >
        <ul
          class="flex flex-wrap gap-2"
          role="list"
        >
          <li>
            <button
              class="px-2 py-1 text-sm border rounded-full bg-gray-100"
            >
              Analyze stack
            </button>
          </li>
          <li>
            <button
              class="px-2 py-1 text-sm border rounded-full bg-gray-100"
            >
              Check risks
            </button>
          </li>
        </ul>
      </div>
    </article>
  </section>
</DocumentFragment>
`;
